st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -144,6 +149,9 @@ def connect(self):
                  '_username': self.username, '_timeout': self.timeout}
         self.logger.debug('Connecting to server', extra=extra)
 
+        if self.bastion_host and self._bastion_connect():
+            conninfo['sock'] = self.bastion_socket
```

Comment:
It looks like fabric does the same thing. https://github.com/fabric/fabric/commit/c2a9bbfd50f560df6c6f9675603fb405c4071cad

In any case, I'd still like some testing with non localhosts as end hosts. Also, since we set environment variables like ST2_AUTH_TOKEN and user provided ones, it would be good to validate those are available in the end host with the bastion setup.